File: backend/internal/vedika/entity/claim_detail.go
// Package entity contains domain models for the Vedika module.
package entity

// ==================== Claim Detail Types ====================

// ClaimDetailFull contains comprehensive claim data for detail view.
// This includes all 23 categories of data from the SIMRS.
type ClaimDetailFull struct {
	// Basic Info
	NoRawat string `json:"no_rawat"`
	Jenis   string `json:"jenis"` // ralan or ranap

	// Core Data
	Patient      PatientInfo      `json:"patient"`
	Registration RegistrationInfo `json:"registration"`
	SEP          *SEPInfo         `json:"sep,omitempty"`
	PRB          *PRBInfo         `json:"prb,omitempty"`

	// Medical Resume
	Resume    *ResumeInfo `json:"resume,omitempty"`
	DPJPRanap []DPJPItem  `json:"dpjp_ranap,omitempty"`

	// Medical Data
	Diagnoses  []DiagnosisItemFull `json:"diagnoses"`
	Procedures []ProcedureItemFull `json:"procedures"`

	// Examinations (SOAP)
	Examinations []ExaminationItem `json:"examinations,omitempty"`

	// Treatments
	TreatmentsRalanDr []TreatmentItem `json:"treatments_ralan_dr,omitempty"`
	TreatmentsRalanPr []TreatmentItem `json:"treatments_ralan_pr,omitempty"`
	TreatmentsRanapDr []TreatmentItem `json:"treatments_ranap_dr,omitempty"`
	TreatmentsRanapPr []TreatmentItem `json:"treatments_ranap_pr,omitempty"`

	// Room Info (Ranap)
	Rooms []RoomInfo `json:"rooms,omitempty"`

	// Surgery
	Surgeries []SurgeryInfo `json:"surgeries,omitempty"`

	// Radiology
	Radiology []RadiologyInfo `json:"radiology,omitempty"`

	// Laboratory
	Laboratory []LabInfo `json:"laboratory,omitempty"`

	// Medicines
	MedicinesGiven   []MedicineInfo `json:"medicines_given,omitempty"`
	MedicinesSurgery []MedicineInfo `json:"medicines_surgery,omitempty"`
	MedicinesHome    []MedicineInfo `json:"medicines_home,omitempty"`

	// Documents
	Documents []DocumentInfoFull `json:"documents,omitempty"`

	// Billing
	Billing []BillingItem `json:"billing,omitempty"`

	// Status
	Status ClaimStatus `json:"status"`
}

// ==================== Patient & Registration ====================

// PatientInfo contains patient demographic data.
type PatientInfo struct {
	NoRkmMedis   string `json:"no_rkm_medis"`
	NamaPasien   string `json:"nama_pasien"`
	NoKTP        string `json:"no_ktp,omitempty"`
	JenisKelamin string `json:"jenis_kelamin"`
	TglLahir     string `json:"tgl_lahir"`
	Umur         string `json:"umur"`
	Alamat       string `json:"alamat"`
	Kelurahan    string `json:"kelurahan,omitempty"`
	Kecamatan    string `json:"kecamatan,omitempty"`
	Kabupaten    string `json:"kabupaten,omitempty"`
	NoTelp       string `json:"no_tlp,omitempty"`
	Pekerjaan    string `json:"pekerjaan,omitempty"`
}

// RegistrationInfo contains registration/visit data.
type RegistrationInfo struct {
	NoRawat       string `json:"no_rawat"`
	TglRegistrasi string `json:"tgl_registrasi"`
	JamReg        string `json:"jam_reg"`
	NamaDokter    string `json:"nama_dokter"`
	NamaPoli      string `json:"nama_poli"`
	CaraBayar     string `json:"cara_bayar"`
	StatusLanjut  string `json:"status_lanjut"`
	Status        string `json:"status"`
}

// ==================== SEP & BPJS ====================

// SEPInfo contains BPJS SEP data.
type SEPInfo struct {
	NoSEP        string `json:"no_sep"`
	NoKartu      string `json:"no_kartu"`
	TglSEP       string `json:"tgl_sep"`
	TglRujukan   string `json:"tgl_rujukan,omitempty"`
	BatasRujukan string `json:"batas_rujukan,omitempty"` // tglrujukan + 85 days
	KelasRawat   string `json:"kelas_rawat"`
	JnsRawat     string `json:"jns_rawat"` // 1=Ranap, 2=Ralan
	NmDiagnosa   string `json:"nm_diagnosa,omitempty"`
	AsalRujukan  string `json:"asal_rujukan,omitempty"`
	NmRujukan    string `json:"nm_rujukan,omitempty"`
	Catatan      string `json:"catatan,omitempty"`
}

// PRBInfo contains Program Rujuk Balik data.
type PRBInfo struct {
	NoSEP       string `json:"no_sep"`
	NoPRB       string `json:"no_prb"`
	TglMulai    string `json:"tgl_mulai"`
	TglAkhir    string `json:"tgl_akhir"`
	NmDokter    string `json:"nm_dokter,omitempty"`
	Alamat      string `json:"alamat,omitempty"`
	NmDiagnosa  string `json:"nm_diagnosa,omitempty"`
	NmDiagnosa2 string `json:"nm_diagnosa2,omitempty"`
}

// ==================== Resume & DPJP ====================

// ResumeInfo contains medical resume data.
type ResumeInfo struct {
	NoRawat          string `json:"no_rawat"`
	TglPeriksa       string `json:"tgl_periksa,omitempty"`
	KeluhanUtama     string `json:"keluhan_utama"`
	JalannyaPenyakit string `json:"jalannya_penyakit,omitempty"`
	Anamnesa         string `json:"anamnesa,omitempty"`
	PemeriksaanFisik string `json:"pemeriksaan_fisik"`
	DiagnosaAkhir    string `json:"diagnosa_akhir"`
	Terapi           string `json:"terapi"`
	Anjuran          string `json:"anjuran,omitempty"`
	ProgramTerapi    string `json:"program_terapi,omitempty"`
	CaraKeluar       string `json:"cara_keluar,omitempty"`
	KeadaanKeluar    string `json:"keadaan_keluar,omitempty"`
	NmDokter         string `json:"nm_dokter"`
}

// DPJPItem contains DPJP Ranap data.
type DPJPItem struct {
	TglDPJP  string `json:"tgl_dpjp"`
	NmDokter string `json:"nm_dokter"`
}

// ==================== Diagnoses & Procedures ====================

// DiagnosisItemFull contains diagnosis data with disease name.
type DiagnosisItemFull struct {
	KodePenyakit string `json:"kode_penyakit"`
	NamaPenyakit string `json:"nama_penyakit"`
	StatusDx     string `json:"status_dx"` // Ralan/Ranap
	Prioritas    int    `json:"prioritas"`
}

// ProcedureItemFull contains procedure data with description.
type ProcedureItemFull struct {
	Kode       string `json:"kode"`
	Deskripsi  string `json:"deskripsi"`
	Prioritas  int    `json:"prioritas"`
	StatusProc string `json:"status,omitempty"` // Ralan/Ranap
}

// ==================== Examinations ====================

// ExaminationItem contains SOAP examination data.
type ExaminationItem struct {
	TglPerawatan string `json:"tgl_perawatan"`
	JamRawat     string `json:"jam_rawat"`
	Suhu         string `json:"suhu_tubuh,omitempty"`
	Tensi        string `json:"tensi,omitempty"`
	Nadi         string `json:"nadi,omitempty"`
	Respirasi    string `json:"respirasi,omitempty"`
	Tinggi       string `json:"tinggi,omitempty"`
	Berat        string `json:"berat,omitempty"`
	SpO2         string `json:"spo2,omitempty"`
	GCS          string `json:"gcs,omitempty"`
	Kesadaran    string `json:"kesadaran,omitempty"`
	Keluhan      string `json:"keluhan,omitempty"`
	Pemeriksaan  string `json:"pemeriksaan,omitempty"`
	Penilaian    string `json:"penilaian,omitempty"`
	Rtl          string `json:"rtl,omitempty"` // Rencana Tindak Lanjut
	Instruksi    string `json:"instruksi,omitempty"`
	Evaluasi     string `json:"evaluasi,omitempty"`
	NmDokter     string `json:"nm_dokter,omitempty"`
	NmPerawat    string `json:"nm_perawat,omitempty"`
}

// ==================== Treatments ====================

// TreatmentItem contains treatment/procedure data.
type TreatmentItem struct {
	TglPerawatan string  `json:"tgl_perawatan"`
	JamRawat     string  `json:"jam_rawat"`
	NmPerawatan  string  `json:"nm_perawatan"`
	NmDokter     string  `json:"nm_dokter,omitempty"`
	NmPerawat    string  `json:"nm_perawat,omitempty"`
	Biaya        float64 `json:"biaya"`
}

// ==================== Room (Ranap) ====================

// RoomInfo contains room/ward data for inpatient.
type RoomInfo struct {
	TglMasuk     string  `json:"tgl_masuk"`
	JamMasuk     string  `json:"jam_masuk"`
	TglKeluar    string  `json:"tgl_keluar,omitempty"`
	JamKeluar    string  `json:"jam_keluar,omitempty"`
	KdKamar      string  `json:"kd_kamar"`
	NmBangsal    string  `json:"nm_bangsal"`
	Tarif        float64 `json:"tarif"`
	LamaMenginap int     `json:"lama_menginap"`
	TotalBiaya   float64 `json:"total_biaya"`
}

// ==================== Surgery ====================

// SurgeryInfo contains surgery/operation data.
type SurgeryInfo struct {
	TglOperasi  string  `json:"tgl_operasi"`
	JnsOperasi  string  `json:"jns_operasi"` // Besar/Sedang/Kecil
	NmPerawatan string  `json:"nm_perawatan"`
	Operator1   string  `json:"operator1,omitempty"`
	Operator2   string  `json:"operator2,omitempty"`
	Operator3   string  `json:"operator3,omitempty"`
	Asisten1    string  `json:"asisten1,omitempty"`
	Asisten2    string  `json:"asisten2,omitempty"`
	DokterAnak  string  `json:"dokter_anak,omitempty"`
	Dokter      string  `json:"dokter,omitempty"`
	Biaya       float64 `json:"biaya,omitempty"`
}

// ==================== Radiology ====================

// RadiologyInfo contains radiology exam data.
type RadiologyInfo struct {
	TglPeriksa  string  `json:"tgl_periksa"`
	JamPeriksa  string  `json:"jam_periksa"`
	NmPerawatan string  `json:"nm_perawatan"`
	NmDokter    string  `json:"nm_dokter"`
	NmPetugas   string  `json:"nm_petugas,omitempty"`
	Hasil       string  `json:"hasil,omitempty"`
	Biaya       float64 `json:"biaya"`
}

// ==================== Laboratory ====================

// LabInfo contains laboratory exam data.
type LabInfo struct {
	TglPeriksa  string      `json:"tgl_periksa"`
	JamPeriksa  string      `json:"jam_periksa"`
	NmPerawatan string      `json:"nm_perawatan"`
	NmDokter    string      `json:"nm_dokter,omitempty"`
	Biaya       float64     `json:"biaya"`
	Details     []LabDetail `json:"details,omitempty"`
}

// LabDetail contains individual lab test results.
type LabDetail struct {
	Pemeriksaan  string `json:"pemeriksaan"`
	Hasil        string `json:"hasil"`
	Satuan       string `json:"satuan,omitempty"`
	NilaiRujukan string `json:"nilai_rujukan,omitempty"`
	Keterangan   string `json:"keterangan,omitempty"`
}

// ==================== Medicines ====================

// MedicineInfo contains medicine/drug data.
type MedicineInfo struct {
	TglPerawatan string  `json:"tgl_perawatan"`
	JamPerawatan string  `json:"jam_perawatan,omitempty"`
	NamaObat     string  `json:"nama_obat"`
	Jumlah       float64 `json:"jumlah"`
	Satuan       string  `json:"satuan,omitempty"`
	Dosis        string  `json:"dosis,omitempty"`
	Aturan       string  `json:"aturan_pakai,omitempty"`
	TotalBiaya   float64 `json:"total_biaya,omitempty"`
}

// ==================== Documents ====================

// DocumentInfoFull contains uploaded document metadata.
type DocumentInfoFull struct {
	Kode       string `json:"kode"`
	NamaBerkas string `json:"nama_berkas"`
	Lokasi     string `json:"lokasi"`
	FilePath   string `json:"file_path,omitempty"` // URL for download
}

// ==================== Billing ====================

// BillingItem contains billing line item.
type BillingItem struct {
	No          string  `json:"no"`
	NmPerawatan string  `json:"nm_perawatan"`
	Pemisah     string  `json:"pemisah,omitempty"`
	Biaya       float64 `json:"biaya"`
	Jumlah      float64 `json:"jumlah"`
	Tambahan    float64 `json:"tambahan,omitempty"`
	TotalBiaya  float64 `json:"total_biaya"`
}

File: Frontend/src/types/claimDetail.ts
// Comprehensive Claim Detail Types - matches backend ClaimDetailFull
// All 23 data categories from SIMRS

export type JenisLayanan = 'ralan' | 'ranap' | 'Ralan' | 'Ranap';
export type ClaimStatus = 'RENCANA' | 'PENGAJUAN' | 'PERBAIKAN' | 'LENGKAP' | 'SETUJU';

// ==================== Patient & Registration ====================

export interface PatientInfo {
    no_rkm_medis: string;
    nama_pasien: string;
    no_ktp?: string;
    jenis_kelamin: string;
    tgl_lahir: string;
    umur?: string;
    alamat: string;
    kelurahan?: string;
    kecamatan?: string;
    kabupaten?: string;
    no_tlp?: string;
    pekerjaan?: string;
}

export interface RegistrationInfo {
    no_rawat: string;
    tgl_registrasi: string;
    jam_reg: string;
    nama_dokter: string;
    nama_poli: string;
    cara_bayar: string;
    status_lanjut: string;
    status: string;
}

// ==================== SEP & BPJS ====================

export interface SEPInfo {
    no_sep: string;
    no_kartu: string;
    tgl_sep: string;
    tgl_rujukan?: string;
    batas_rujukan?: string;
    kelas_rawat: string;
    jns_rawat: string;
    nm_diagnosa?: string;
    asal_rujukan?: string;
    nm_rujukan?: string;
    catatan?: string;
}

export interface PRBInfo {
    no_sep: string;
    no_prb: string;
    tgl_mulai: string;
    tgl_akhir: string;
    nm_dokter?: string;
    alamat?: string;
    nm_diagnosa?: string;
    nm_diagnosa2?: string;
}

// ==================== Resume & DPJP ====================

export interface ResumeInfo {
    no_rawat: string;
    tgl_periksa?: string;
    keluhan_utama: string;
    jalannya_penyakit?: string;
    anamnesa?: string;
    pemeriksaan_fisik: string;
    diagnosa_akhir: string;
    terapi: string;
    anjuran?: string;
    program_terapi?: string;
    cara_keluar?: string;
    keadaan_keluar?: string;
    nm_dokter: string;
}

export interface DPJPItem {
    tgl_dpjp: string;
    nm_dokter: string;
}

// ==================== Diagnoses & Procedures ====================

export interface DiagnosisItemFull {
    kode_penyakit: string;
    nama_penyakit: string;
    status_dx: string;
    prioritas: number;
}

export interface ProcedureItemFull {
    kode: string;
    deskripsi: string;
    prioritas: number;
    status?: string;
}

// ==================== Examinations ====================

export interface ExaminationItem {
    tgl_perawatan: string;
    jam_rawat: string;
    suhu_tubuh?: string;
    tensi?: string;
    nadi?: string;
    respirasi?: string;
    tinggi?: string;
    berat?: string;
    spo2?: string;
    gcs?: string;
    kesadaran?: string;
    keluhan?: string;
    pemeriksaan?: string;
    penilaian?: string;
    rtl?: string;
    instruksi?: string;
    evaluasi?: string;
    nm_dokter?: string;
    nm_perawat?: string;
}

// ==================== Treatments ====================

export interface TreatmentItem {
    tgl_perawatan: string;
    jam_rawat: string;
    nm_perawatan: string;
    nm_dokter?: string;
    nm_perawat?: string;
    biaya: number;
}

// ==================== Room (Ranap) ====================

export interface RoomInfo {
    tgl_masuk: string;
    jam_masuk: string;
    tgl_keluar?: string;
    jam_keluar?: string;
    kd_kamar: string;
    nm_bangsal: string;
    tarif?: number;
    lama_menginap: number;
    total_biaya: number;
}

// ==================== Surgery ====================

export interface SurgeryInfo {
    tgl_operasi: string;
    jns_operasi: string;
    nm_perawatan: string;
    operator1?: string;
    operator2?: string;
    operator3?: string;
    asisten1?: string;
    asisten2?: string;
    dokter_anak?: string;
    dokter?: string;
    biaya?: number;
}

// ==================== Radiology ====================

export interface RadiologyInfo {
    tgl_periksa: string;
    jam_periksa: string;
    nm_perawatan: string;
    nm_dokter: string;
    nm_petugas?: string;
    hasil?: string;
    biaya: number;
}

// ==================== Laboratory ====================

export interface LabDetail {
    pemeriksaan: string;
    hasil: string;
    satuan?: string;
    nilai_rujukan?: string;
    keterangan?: string;
}

export interface LabInfo {
    tgl_periksa: string;
    jam_periksa: string;
    nm_perawatan: string;
    nm_dokter?: string;
    biaya: number;
    details?: LabDetail[];
}

// ==================== Medicines ====================

export interface MedicineInfo {
    tgl_perawatan: string;
    jam_perawatan?: string;
    nama_obat: string;
    jumlah: number;
    satuan?: string;
    dosis?: string;
    aturan_pakai?: string;
    total_biaya?: number;
}

// ==================== Documents ====================

export interface DocumentInfoFull {
    kode: string;
    nama_berkas: string;
    lokasi: string;
    file_path?: string;
}

// ==================== Billing ====================

export interface BillingItem {
    no: string;
    nm_perawatan: string;
    pemisah?: string;
    biaya: number;
    jumlah: number;
    tambahan?: number;
    total_biaya: number;
}

// ==================== Main ClaimDetailFull ====================

export interface ClaimDetailFull {
    no_rawat: string;
    jenis: string;

    // Core Data
    patient: PatientInfo;
    registration: RegistrationInfo;
    sep?: SEPInfo;
    prb?: PRBInfo;

    // Medical Resume
    resume?: ResumeInfo;
    dpjp_ranap?: DPJPItem[];

    // Medical Data
    diagnoses: DiagnosisItemFull[];
    procedures: ProcedureItemFull[];

    // Examinations (SOAP)
    examinations?: ExaminationItem[];

    // Treatments
    treatments_ralan_dr?: TreatmentItem[];
    treatments_ralan_pr?: TreatmentItem[];
    treatments_ranap_dr?: TreatmentItem[];
    treatments_ranap_pr?: TreatmentItem[];

    // Room Info (Ranap)
    rooms?: RoomInfo[];

    // Surgery
    surgeries?: SurgeryInfo[];

    // Radiology
    radiology?: RadiologyInfo[];

    // Laboratory
    laboratory?: LabInfo[];

    // Medicines
    medicines_given?: MedicineInfo[];
    medicines_surgery?: MedicineInfo[];
    medicines_home?: MedicineInfo[];

    // Documents
    documents?: DocumentInfoFull[];

    // Billing
    billing?: BillingItem[];

    // Status
    status: ClaimStatus;
}

export interface ClaimDetailFullResponse {
    success: boolean;
    data: ClaimDetailFull;
}
